responder/msplanner/msplanner.go

```diff
@@ -0,0 +1,214 @@
+		return openDataError(err)
+	}
+
+	post := models.NewPost()
+	body := models.NewItemBody()
+	contentType := models.TEXT_BODYTYPE
+	body.SetContentType(&contentType)
+	body.SetContent(&comment)
+	post.SetBody(body)
+
+	// If conversation thread exists, add a new reply
+	if task.GetConversationThreadId() != nil {
+		replyBody := reply.NewReplyPostRequestBody()
+		replyBody.SetPost(post)
+
+		err = c.client.GroupsById(c.groupID).ThreadsById(*task.GetConversationThreadId()).Reply().Post(replyBody)
+		return openDataError(err)
+	}
+
+	// Create a new conversation thread for the task
+	convBody := models.NewConversationThread()
+	topic := fmt.Sprintf("Conversation thread topic for taskID: %s", taskID)
+	convBody.SetTopic(&topic)
+	convBody.SetPosts([]models.Postable{post})
+
+	result, err := c.client.GroupsById(c.groupID).Threads().Post(convBody)
+	if err != nil {
+		return openDataError(err)
+	}
+
+	// Link the created conversation thread to the task
+	etag := *task.GetAdditionalData()["@odata.etag"].(*string)
+	headers := map[string]string{"If-Match": etag}
+	patchConfig := item.PlannerTaskItemRequestBuilderPatchRequestConfiguration{Headers: headers}
+
+	requestBody := models.NewPlannerTask()
+	requestBody.SetConversationThreadId(result.GetId())
+	err = c.client.Planner().TasksById(taskID).PatchWithRequestConfigurationAndResponseHandler(requestBody, &patchConfig, nil)
+	return openDataError(err)
+}
+
+func (c MSPlannerClient) GetComments(taskID string) ([]api.Comment, error) {
+	task, err := c.client.Planner().TasksById(taskID).Get()
+	if err != nil {
+		return nil, openDataError(err)
+	}
+
+	var comments []api.Comment
+	if task.GetConversationThreadId() == nil {
+		return comments, nil
+	}
+
+	conversations, err := c.client.GroupsById(c.groupID).ThreadsById(*task.GetConversationThreadId()).Posts().Get()
+	if err != nil {
+		return nil, openDataError(err)
+	}
+
+	for _, conv := range conversations.GetValue() {
+		comments = append(comments, api.Comment{
+			Body:      *conv.GetBody().GetContent(),
+			CreatedBy: *conv.GetFrom().GetEmailAddress().GetName(),
+			CreatedAt: *conv.GetCreatedDateTime(),
+		})
+	}
+
+	return comments, nil
+}
+
+func (c MSPlannerClient) GetConfig() (map[string]PlanConfig, error) {
+	var priorities []string
+	for k := range taskPriorities {
+		priorities = append(priorities, k)
+	}
+
+	config := make(map[string]PlanConfig)
+	result, err := c.client.GroupsById(c.groupID).Planner().Plans().Get()
+	if err != nil {
+		return config, openDataError(err)
+	}
+
+	for _, plan := range result.GetValue() {
+		var buckets []PlanBucket
+		for _, bucket := range plan.GetBuckets() {
+			buckets = append(buckets, PlanBucket{
+				ID:   *bucket.GetId(),
+				Name: *bucket.GetName(),
+			})
+		}
+
+		config[*plan.GetTitle()] = PlanConfig{
+			ID:         *plan.GetId(),
+			Name:       *plan.GetTitle(),
+			Buckets:    buckets,
+			Priorities: priorities,
+		}
+	}
+
+	return config, nil
+}
+
+func openDataError(err error) error {
```

Comment:
just curious why do we need to cater for it specially

Reply:
The microsoft graph API returns specific `openData` errors where you have two extract multiple things from the error. If I just log the error as is, we'll get a `error bad status code received`